crypto/x509/x_req.c

```diff
@@ -53,13 +53,16 @@ static int req_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
     switch (operation) {
     case ASN1_OP_D2I_PRE:
         ASN1_OCTET_STRING_free(ret->distinguishing_id);
+        OPENSSL_free(ret->propq);
```

Comment:
I dont think the one in d2i_pre should be done.
i.e We would normally create an empty object that has a propq and libctx set and then call d2i.
So clearing it would free the item we just set.  I deliberately did not do this in the other objects for X509 and X509_CRL

Reply:
OK, I see. I got confused by the `ASN1_OCTET_STRING_free(ret->distinguishing_id)` and the fact this is `ASN1_OP_D2I_PRE`, which I thought is called before the values are set. But I will remove these two lines.

Reply:
Thanks for finding the leak..